protected/humhub/modules/user/controllers/AccountController.php

```diff
@@ -332,7 +334,60 @@ public function actionChangeEmailValidate()
         return $this->render('changeEmailValidate', ['newEmail' => $email]);
     }
 
+	/**
+	 * Change users current username
+	 */
+	public function actionChangeUsername()
+	{
+		if (!Yii::$app->user->canChangeUsername()) {
+			throw new HttpException(500, 'Change Username is not allowed');
+		}
+
+		$model = new \humhub\modules\user\models\forms\AccountChangeUsername;
+
+		if ($model->load(Yii::$app->request->post()) && $model->validate() && $model->sendChangeUsername()) {
+            $this->view->warn(Yii::t('UserModule.controllers_AccountController', 'Changing the username can make some links unusable, for example old links to the profile.'));
```

Comment:
We rather should fire a `$this->view->success()` success here, the warning within the form view should be sufficient.